game-core/src/main/java/games/strategy/triplea/ResourceLoader.java

```diff
@@ -308,7 +308,7 @@ public boolean hasPath(final String path) {
       return Optional.empty();
     }
     try {
-      return Optional.of(ImageIO.read(url));
+      return Optional.ofNullable(ImageIO.read(url));
```

Comment:
Hmm I'd suggest logging some kind of error message if `ImageIO.read` returns `null`.
Otherwise map makers might not realize that an invalid image was attempted to get loaded

Reply:
Hello, that's my first PR so i'm not entirely sure how to proceed from here :)
About the logs, I guess you are right, the caller should handle the null but an error log should indicate exactly what happened (maybe that image was invalid?).

Reply:
I confess I really don't know exactly in what circumstances this might occur.

Reply:
From the javadocs of [`ImageIO#read`](https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/javax/imageio/ImageIO.html#read(java.net.URL)):
> If no registered ImageReader claims to be able to read the resulting stream, null is returned.

This seems to be the case if the provided Image format is not supported.

So my suggestion would be to add an if-check that checks for null, and in case it is actually null the logger is called something like this:
```java
log.severe("Unsupported Image Format: " + url);
```

Maybe you can come up with a better wording for the error message